noir-projects/aztec-nr/aztec/src/macros/mod.nr

```diff
@@ -147,27 +167,32 @@ comptime fn generate_compute_note_hash_and_optionally_a_nullifier() -> Quoted {
         let if_statements = if_statements_list.join(quote {});
 
         quote {
-            $if_statements
-            else {
-                panic(f"Unknown note type ID: {note_type_id}")
+            unconstrained fn compute_note_hash_and_optionally_a_nullifier(
```

Comment:
Had to do this change because if we returned a body from the if and then return the final quote from the function later we would get warnings in cases where there are no notes as the function args would be unused.